src/data/weather.ts

```diff
@@ -152,17 +152,16 @@ export const getWeatherUnit = (
   hass: HomeAssistant,
   measure: string
 ): string => {
-  const lengthUnit = hass.config.unit_system.length || "";
   switch (measure) {
     case "pressure":
-      return lengthUnit === "km" ? "hPa" : "inHg";
+      return hass.config.unit_system.pressure || "";
     case "wind_speed":
-      return `${lengthUnit}/h`;
+      return hass.config.unit_system.wind_speed || "";
     case "visibility":
     case "length":
-      return lengthUnit;
+      return hass.config.unit_system.length || "";
```

Comment:
Remove the case for `pressure`, `wind_speed`, and `length` as they can be handled by the `default` now.